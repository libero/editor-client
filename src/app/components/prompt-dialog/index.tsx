import React from 'react';
import { Dialog, DialogActions, DialogContent, DialogContentText, DialogTitle, Typography } from '@material-ui/core';
import { ActionButton, VariantType } from '../action-button';

interface PromptDialogProps {
  isOpen: boolean;
  title?: string;
  message: string;
  onAccept: () => void;
  onReject: () => void;
  acceptLabel?: string;
  rejectLabel?: string;
  acceptVariant?: VariantType;
  rejectVariant?: VariantType;
}

const defaultProps = {
  acceptLabel: 'OK',
  rejectLabel: 'Cancel',
  acceptVariant: 'primaryContained' as VariantType,
  rejectVariant: 'secondaryOutlined' as VariantType
};

export const PromptDialog: React.FC<PromptDialogProps> = (props) => {
  return (
    <Dialog open={props.isOpen} aria-labelledby="alert-dialog-title" aria-describedby="alert-dialog-description">
      <DialogTitle id="alert-dialog-title">
        <Typography variant="h2">{props.title}</Typography>
      </DialogTitle>
      <DialogContent>
        <DialogContentText id="alert-dialog-description">{props.message}</DialogContentText>
      </DialogContent>
      <DialogActions>
        <ActionButton
          onClick={props.onReject}
          variant={props.rejectVariant || defaultProps.rejectVariant}
          title={props.rejectLabel || defaultProps.rejectLabel}
        />
        <ActionButton
          onClick={props.onAccept}
          variant={props.acceptVariant || defaultProps.acceptVariant}
          title={props.acceptLabel || defaultProps.acceptLabel}
        />
      </DialogActions>
    </Dialog>
  );
};
