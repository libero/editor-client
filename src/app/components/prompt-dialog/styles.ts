import { makeStyles } from '@material-ui/core/styles';

export const usePromptDialogStyles = makeStyles((theme) => ({
  dialogTitle: {
    padding: theme.spacing(3, 3, 1, 3)
  },
  actionPanel: {
    padding: 0,
    marginBottom: theme.spacing(3),
    marginRight: theme.spacing(3),
    '& > *:last-child': {
      marginLeft: theme.spacing(2)
    }
  }
}));
