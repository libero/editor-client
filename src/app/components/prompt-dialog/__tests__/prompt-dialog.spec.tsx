import React from 'react';
import { create } from 'react-test-renderer';
import { PromptDialog } from 'app/components/prompt-dialog/index';

jest.mock('@material-ui/core', () => ({
  Dialog: ({ children }) => <div data-cmp="dialog">{children}</div>,
  DialogTitle: ({ children }) => <div data-cmp="dialog-title">{children}</div>,
  DialogActions: ({ children }) => <div data-cmp="dialog-actions">{children}</div>,
  DialogContent: ({ children }) => <div data-cmp="dialog-content">{children}</div>,
  DialogContentText: ({ children }) => <div data-cmp="dialog-content-text">{children}</div>,
  Content: ({ children }) => <div data-cmp="dialog-content">{children}</div>,
  Typography: () => <div data-cmp="typography"></div>,
  Button: () => <div data-cmp="Button"></div>
}));

describe('Prompt Dialog', () => {
  it('renders prompt dialog', () => {
    const wrapper = create(
      <PromptDialog
        isOpen={true}
        message="testMessage"
        title="testTitle"
        onReject={jest.fn()}
        onAccept={jest.fn()}
        acceptLabel="Okido"
        rejectLabel="Nope"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders prompt with default params', () => {
    const wrapper = create(
      <PromptDialog isOpen={true} message="testMessage" title="testTitle" onReject={jest.fn()} onAccept={jest.fn()} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
