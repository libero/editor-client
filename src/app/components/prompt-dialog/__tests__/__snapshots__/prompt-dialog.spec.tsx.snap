// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prompt Dialog renders prompt dialog 1`] = `
<div
  data-cmp="dialog"
>
  <div
    data-cmp="dialog-title"
  >
    <div
      data-cmp="typography"
    />
  </div>
  <div
    data-cmp="dialog-content"
  >
    <div
      data-cmp="dialog-content-text"
    >
      testMessage
    </div>
  </div>
  <div
    data-cmp="dialog-actions"
  >
    <div
      data-cmp="Button"
    />
    <div
      data-cmp="Button"
    />
  </div>
</div>
`;

exports[`Prompt Dialog renders prompt with default params 1`] = `
<div
  data-cmp="dialog"
>
  <div
    data-cmp="dialog-title"
  >
    <div
      data-cmp="typography"
    />
  </div>
  <div
    data-cmp="dialog-content"
  >
    <div
      data-cmp="dialog-content-text"
    >
      testMessage
    </div>
  </div>
  <div
    data-cmp="dialog-actions"
  >
    <div
      data-cmp="Button"
    />
    <div
      data-cmp="Button"
    />
  </div>
</div>
`;
